Cargo.toml

```diff
@@ -17,14 +17,14 @@ anyhow = "1.0.75"
 async-trait = "0.1"
 clap = { version = "3.1.15", features = ["derive", "env"] }
 serde = "1.0"
-spin-app = { git = "https://github.com/fermyon/spin", tag = "v2.3.1" }
-spin-core = { git = "https://github.com/fermyon/spin", tag = "v2.3.1" }
-spin-trigger = { git = "https://github.com/fermyon/spin", tag = "v2.3.1" }
+spin-app = { git = "https://github.com/fermyon/spin" }
```

Comment:
nit: leave a TODO to pin to the Spin `2.4` release for these dependencies.

Reply:
Do we need to pin to 2.4?

I pinned to 2.3.1 earlier so this can be imported in the containerd shim.

Reply:
That's a good point and probably necessitates implementing this in terms of what's available in 2.3.1.
 
I only noticed the upgrade to the various new traits used below.

Reply:
These can definitely be implemented pinned to `2.3.1` but I upgraded as I was following the example of the `http-trigger`. Do we think pinning it to 2.3.1 is a necessity or will the shim be updated to `2.4.0` once that is released?

Reply:
We'll bump everything once 2.4 is out.